python/tskit/trees.py

```diff
@@ -10476,6 +10476,45 @@ def ld_matrix(
             mode=mode,
         )
 
+    def individual_nodes(self):
+        """
+        Return an array of node IDs for each individual in the tree sequence.
+        Errors if the tree sequence has no individuals, if any sample doesn't
+        have an individual, if individuals have nodes that are both samples
+        and non-samples, or if an individual has no samples.
+
+        :return: Array of shape (num_individuals, max_ploidy) containing node IDs.
+            Values of -1 indicate unused slots for individuals with ploidy
+            less than the maximum.
+        :rtype: numpy.ndarray (dtype=np.int32)
+        """
+        if self.num_individuals == 0:
+            raise ValueError("Tree sequence has no individuals")
+
+        max_ploidy = 0
+        for i in range(self.num_individuals):
+            ind = self.individual(i)
+            max_ploidy = max(max_ploidy, len(ind.nodes))
+        if max_ploidy == 0:
+            raise ValueError("No nodes refer to individuals")
+
+        # Initialize output array with -1 (indicating no node)
+        result = np.full((self.num_individuals, max_ploidy), -1, dtype=np.int32)
+
+        for i in range(self.num_individuals):
+            ind = self.individual(i)
+            if len(ind.nodes) == 0:
```

Comment:
I don't think we should check for sample status here, we're just returning the actual nodes. There's no reason that an individual couldn't have mixed sample status or not be a sample at all I think.

Reply:
Agree - that'd be weird, but we can think of use cases - what if we have sequence data for a parent and only sparse data for a child, such that we can impute one of the sets of chromosomes for the child but not the other.